Makefile

```diff
@@ -35,29 +35,70 @@ else
     LDLIBS += -lgomp
     $(info NICE Compiling with OpenMP support)
   else
-    $(warning OOPS Compiling without OpenMP support)
+    ifneq ($(OS), Windows_NT)
+      $(warning OOPS Compiling without OpenMP support)
+    endif
   endif
 endif
 
+ifeq ($(OS), Windows_NT)
+  CC := cl 
+  CFLAGS := /Iplatform\windows /Zi /nologo /W3 /WX- /diagnostics:column /sdl /Ox /Oi /Ot /GL /D _DEBUG /D _CONSOLE /D _UNICODE /D UNICODE /Gm- /EHsc /MD /GS /Gy /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /permissive- \
```

Comment:
@rosslwheeler the issue here is /fp:precise combined with /Ox. #19 is about this. What we want is /fp:fast with /Ox or /O2 to work without producing a nan.

Reply:
@azret - open to suggestions on the optimizations - don't have a strong opinion on the compiler options. Let me know what you'd like to try here. Thanks.

Reply:
It's not failing - I think the issue per @azret was that we can tweak the optimizations better (faster). I'll need to look up the syntax to see what @azret is referring to above and give it a try.

Reply:
Basically, `train_gpt2` fails after the first step when compiled with `/fp:fast` (or `-ffast_math` on gcc). This should omit that optimization just for the function that causes the failure, while keeping the speedup for everything else.

Reply:
By "not failing" you mean it runs some training steps fine, even when compiled with `/fp:fast /Ox`?